File: atcoder/abc395a_strictly_increasing.py
def strictly_increasing(a):
    return all(a[i] < a[i + 1] for i in range(len(a) - 1))

if __name__ == "__main__":
    n = int(input())
    a = list(map(int, input().split()))
    print("Yes" if strictly_increasing(a) else "No")

File: atcoder/abc395a_strictly_increasing.rs
use std::io;

fn strictly_increasing(a: &[i32]) -> bool {
    a.windows(2).all(|w| w[0] < w[1])
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let _n: usize = input.trim().parse().unwrap();
    
    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let a: Vec<i32> = input.trim().split_whitespace()
                           .map(|x| x.parse().unwrap())
                           .collect();
    
    println!("{}", if strictly_increasing(&a) { "Yes" } else { "No" });
}
